pkg/rtc/room.go

```diff
@@ -421,14 +423,14 @@ func (r *Room) onParticipantMetadataUpdate(p types.Participant) {
 	}
 }
 
-func (r *Room) onDataPacket(source types.Participant, dp *livekit.DataPacket) {
+func (r *Room) SendDataPacket(dp *livekit.DataPacket, source *types.Participant) {
```

Comment:
I'm thinking now that hijacking this function is probably not the best idea.. a couple of reasons:

* it's strange to take in a source parameter when the function is sending data packets. you would almost always leave this as nil, if used as a backend API
* ideally it would take in a `UserPacket`, and not a DataPacket. We do not want to allow Room to have a public interface to send activespeaker updates, for example.

I would suggest doing this:

* creating a SendDataPacket function, that takes in a UserPacket
* in this function, wrap it in a DataPacket and call the private func `onDataPacket(nil, dp)`
* leave the callback handlers the way it is.

Reply:
Thanks for the suggestions ✌🏻
I was wondering if it was a good idea to modify `onDataPacket` or make another function (but it's clear now).

So for `onDataPacket(nil, dp)`, optionals come first (in go) am I correct?

Reply:
typically it still comes last. But with onDataPacket, the data packet will always come from a participant (since this is coming from within a room)

Reply:
Yes I noticed that now 😅

Reply:
> call the private func onDataPacket(nil, dp)

If I make `onDataPacket`'s `source` optional,
```go
func (r *Room) onDataPacket(source *types.Participant, dp *livekit.DataPacket)
```
> leave the callback handlers the way it is.

It's still not compatible with
```go
  participant.OnDataPacket(r.onDataPacket)
```
So I still need to write like this am I correct?
```go
  participant.OnDataPacket(func(p types.Participant, dp *livekit.DataPacket) {
    r.onDataPacket(&p, dp)
  })
```

Reply:
hmm what I'm proposing is to leave onDataPacket alone.. don't change its signature at all.

Why do this? `*types.Participant` ? it's not typically to use a pointer to an interface in Go. You can pass in nil for an interface value.

Reply:
Oh I didn't know can do that in go haha thanks 👏🏻

Reply:
I pushed the changes, it's starting to look better now. Is it necessary to do `source != nil` check ?

Reply:
yeah, you should perform that check.